api-extractor/src/ApiDefinitionReference.ts

```diff
@@ -0,0 +1,195 @@
+  }
+
+  /**
+   * Takes an API reference expression of the form '@scopeName/packageName:exportName.memberName'
+   * and deconstructs it into an IApiDefinitionReference interface object.
+   */
+  public static createFromString(apiReferenceExpr: string,
+    reportError: (message: string) => void): ApiDefinitionReference {
+    if (!apiReferenceExpr || apiReferenceExpr.split(' ').length > 1) {
+      reportError('API reference expression must be of the form: ' +
+        '\'scopeName/packageName:exportName.memberName | display text\'' +
+        'where the \'|\' is required if a display text is provided');
+      return;
+    }
+
+    let scopeName: string = '';
+    let packageName: string = '';
+    let exportName: string = '';
+    let memberName: string = '';
+
+    // E.g. @microsoft/sp-core-library:Guid.equals
+    let parts: string[] = apiReferenceExpr.match(ApiDefinitionReference._packageRegEx);
+    if (parts) {
+      // parts[1] is of the form ‘@microsoft/sp-core-library’ or ‘sp-core-library’
+      const scopePackageName: IScopePackageName = ApiDefinitionReference.parseScopedPackageName(parts[1]);
+      scopeName = scopePackageName.scope;
+      packageName = scopePackageName.package;
+      apiReferenceExpr = parts[2]; // e.g. Guid.equals
+    }
+
+    // E.g. Guid.equals
+    parts = apiReferenceExpr.match(ApiDefinitionReference._memberRegEx);
+    if (parts) {
+      exportName = parts[1]; // e.g. Guid, can never be undefined
+      memberName = parts[2] ? parts[2] : ''; // e.g. equals
+    } else {
+      // the export name is required
+       throw reportError(`Api reference expression contains invalid characters: ${apiReferenceExpr}`);
+    }
+
+    if (!apiReferenceExpr.match(ApiDefinitionReference._exportRegEx)) {
+      throw reportError(`Api reference expression contains invalid characters: ${apiReferenceExpr}`);
+    }
+
+    return ApiDefinitionReference.createFromParts(scopeName, packageName, exportName, memberName);
+  }
+
+  /**
+   * For a scoped NPM package name this separates the scope and package parts.  For example:
+   * parseScopedPackageName('@my-scope/myproject') = { scope: '@my-scope', package: 'myproject' }
+   * parseScopedPackageName('myproject') = { scope: '', package: 'myproject' }
+   */
+  public static parseScopedPackageName(scopedName: string): IScopePackageName {
+    if (scopedName.substr(0, 1) !== '@') {
+      return { scope: '', package: scopedName };
+    }
+
+    const slashIndex: number = scopedName.indexOf('/');
+    if (slashIndex >= 0) {
+      return { scope: scopedName.substr(0, slashIndex), package: scopedName.substr(slashIndex + 1) };
+    } else {
+      throw new Error('Invalid scoped name: ' + scopedName);
+    }
+  }
+
+  /**
+   * Seperates the name property of an ApiPackage into a scope name and the actual 
+   * name of the package.
+   */
+  public static parseApiPackageName(apiPackageName: string): IScopePackageName {
```

Comment:
We chatted about this.  The apiPackage.name is just wrong.  Your next bug will make it easy to get this information, so for now don't worry about resolve local references that specify the package name.